src/main/java/com/spotify/docker/client/DockerClient.java

```diff
@@ -1003,6 +1001,39 @@ void restartContainer(String containerId, int secondsToWaitBeforeRestart)
   void killContainer(String containerId) throws DockerException, InterruptedException;
 
   /**
+   * Kill a docker container.
+   * @param containerId The id of the container to kill.
+   * @param signal Signal used to kill the process.
+   * @throws ContainerNotFoundException
+   *                              if container is not found (404)
+   * @throws DockerException      if a server error occurred (500)
+   * @throws InterruptedException If the thread is interrupted
+   */
+  void killContainer(final String containerId, final TerminationSignal signal)
+      throws DockerException, InterruptedException;
+
+  /**
+   * Supported parameters for {@link #killContainer(String, TerminationSignal)}).
+   */
+  enum TerminationSignal {
+    SIGINT("SIGINT"),
+    SIGQUIT("SIGQUIT"),
+    SIGTERM("SIGTERM"),
+    SIGKILL("SIGKILL"),
+    SIGSTOP("SIGSTOP");
+
```

Comment:
I think it's best to add all signals because else there would be no way to specify a value missing in the enum. (for bash completion, it's different because the user can enter any value.)

If the enum offeres a subset of the available signals, there should be an overloaded `String` variant that allows passing the missing signals. I prefer the first solution.